### PR TITLE
Refactor error handling using ?

### PR DESCRIPTION
**Pull request by gokulsoumya**
Converted the `map_err()` calls to use the `?` operator transparently by implementing `From` for foreign error types.

### CODE REVIEW

**Comment by itsfarseen** [comment]
Hi @gokulsoumya, I don't fully agree with this refactor because, now the error conversions are located far away from the code that does the error conversion.

Previously, if we wanted to find where a `MaxAgeParseError::NonNumericMaxAge` came from, we could just run `ag` or `grep` for `MaxAgeParseError::NonNumericMaxAge` and we can see that it came from `fetch_keys.rs:89`.

But now the search result will show the `From<ParseIntError>` implementation instead, and now we have to track where all does `ParseIntError` occur, and then among those search results, filter out which functions return `Result<_, MaxAgeParseError>`. 

Although the code has become shorter and somewhat cleaner now, it increases the effort required to debug.
Moral of the story is, shorter code is not always better :)

File: breathe-india-backend/src/google_jwt/fetch_keys.rs [hunk 1]
```diff
@@ -82,13 +88,7 @@ fn parse_max_age_value(cache_control_value: &str) -> Result<Duration, MaxAgePars
 
         if String::from("max-age").eq(&key.to_lowercase()) {
             match val {
-                Some(value) => {
-                    return Ok(Duration::from_secs(
-                        value
-                            .parse()
-                            .map_err(|_| MaxAgeParseError::NonNumericMaxAge)?,
```
Comment (itsfarseen) [inline breathe-india-backend/src/google_jwt/fetch_keys.rs:89]: That being said, there are some stuff that need to be changed in error handing code.
For example in this section, 
```rust
foo(
   bar
      .long()
      .function()
      .chain()
      .map_err(|_| ErrType::Err)?
)
```
readability can be improved by extracting out the argument of `foo(<long..stuff>)`. The `?` is easy to miss when its inside arguments. Also the nesting will be reduced when we extract it out.

**Comment by gokulsoumya** [comment]
That's a very valid point and it's better suited here too. I'll extract other useful refactors from this commit and open another PR with some general refactors 👍🏾

File: breathe-india-backend/src/google_jwt/fetch_keys.rs (hunk 1 above)
Comment (gokulsoumya) [reply breathe-india-backend/src/google_jwt/fetch_keys.rs:89]: I'll open a new PR :)

**Pull request closed without merging**